api/app/routers/tickets.py

```diff
@@ -184,136 +280,27 @@ def get_tickets(
-        raise HTTPException(
-            status_code=status.HTTP_404_NOT_FOUND,
-            detail="No insight associated with this ticket",
-        )
-
     if not check_pteam_membership(ticket.dependency.service.pteam, current_user):
         raise NOT_A_PTEAM_MEMBER
 
-    update_request = request.model_dump(exclude_unset=True)
-    _check_request_fields(request, update_request)
-    insight = ticket.insight
-
-    if "description" in update_request.keys() and request.description is not None:
-        insight.description = request.description
-    if (
-        "data_processing_strategy" in update_request.keys()
-        and request.data_processing_strategy is not None
-    ):
-        insight.data_processing_strategy = request.data_processing_strategy
-    if "threat_scenarios" in update_request.keys() and request.threat_scenarios is not None:
-        for threat_scenario in insight.threat_scenarios:
-            persistence.delete_threat_scenario(db, threat_scenario)
-
-        for threat_scenario_request in request.threat_scenarios:
-            threat_scenario_model = models.ThreatScenario(
-                insight_id=str(insight.insight_id),
-                impact_category=threat_scenario_request.impact_category,
-                title=threat_scenario_request.title,
-                description=threat_scenario_request.description,
-            )
-            persistence.create_threat_scenario(db, threat_scenario_model)
-
-    if "affected_objects" in update_request.keys() and request.affected_objects is not None:
-        for affected_object in insight.affected_objects:
-            persistence.delete_affected_object(db, affected_object)
-
-        for affected_object in request.affected_objects:
-            affected_object_model = models.AffectedObject(
-                insight_id=str(insight.insight_id),
-                object_category=affected_object.object_category,
-                name=affected_object.name,
-                description=affected_object.description,
-            )
-            persistence.create_affected_object(db, affected_object_model)
-
-    if "insight_references" in update_request.keys() and request.insight_references is not None:
-        for insight_reference in insight.insight_references:
-            persistence.delete_insight_reference(db, insight_reference)
-
-        for insight_reference_request in request.insight_references:
-            insight_reference_model = models.InsightReference(
-                insight_id=str(insight.insight_id),
-                link_text=insight_reference_request.link_text,
-                url=insight_reference_request.url,
-            )
-            persistence.create_insight_reference(db, insight_reference_model)
-
-    insight.updated_at = datetime.now(timezone.utc)
+    if not (insight := ticket.insight):
+        insight = __handle_create_insight(ticket_id, request, db)
+    else:
+        insight = __handle_update_insight(insight, request, db)
```

Comment:
```suggestion
        __handle_update_insight(insight, request, db)
```
再代入はできれば避けたい。
ここでは同一オブジェクトであり代入不要にみえる。

Reply:
修正しました